Sources/SwiftLexicalLookup/Scopes/GenericParameterScopeSyntax.swift

```diff
@@ -0,0 +1,91 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+/// Scope that introduces generic parameter names and directs
+/// futher lookup to it's `WithGenericParametersScopeSyntax`
+/// parent scope's parent scope (i.e. on return, bypasses names
+/// introduced by it's parent).
+@_spi(Experimental) public protocol GenericParameterScopeSyntax: ScopeSyntax {}
+
+@_spi(Experimental) extension GenericParameterScopeSyntax {
+  /// Returns names matching lookup and bypasses
+  /// `WithGenericParametersScopeSyntax` parent scope in futher lookup.
+  ///
+  /// example:
```

Comment:
Let’s take advantage of markdown formatting. We do this in other places already. I think there are a few more places in this PR that use `example:`.

```suggestion
  /// ### Example
```